clients/fides-js/src/components/tcf/TcfPurposes.tsx

```diff
@@ -130,25 +74,66 @@ const TcfPurposes = ({
     [allPurposesConsent, allPurposesLegint]
   );
 
+  const [activeLegalBasisOption, setActiveLegalBasisOption] = useState(
+    LEGAL_BASIS_OPTIONS[0]
+  );
+  const activeData: {
+    purposes: PurposeRecord[];
+    purposeModelType: keyof EnabledIds;
+    enabledPurposeIds: string[];
+    specialPurposes: TCFSpecialPurposeRecord[];
+    enabledSpecialPurposeIds: string[];
+  } = useMemo(() => {
+    if (activeLegalBasisOption.value === LegalBasisEnum.CONSENT) {
+      return {
+        purposes: uniquePurposes.filter((p) => p.isConsent),
+        purposeModelType: "purposesConsent",
+        enabledPurposeIds: enabledPurposeConsentIds,
+        // TODO: do we need to filter special purposes too?
```

Comment:
yes—we should make a follow up ticket for this. dunno if it can make it into this sprint though, would need both FE and BE work

Reply:
https://ethyca.atlassian.net/browse/PROD-1318

Reply:
done, only possible thanks to @pattisdr 's quick backend work! [1543e88](https://github.com/ethyca/fides/pull/4376/commits/1543e88a6603656739de3bbe18011ce54f0c8a54)